core/src/main/java/org/jdbi/v3/locator/ClasspathSqlLocator.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.locator;
+
+import static java.util.Collections.synchronizedMap;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Map;
+import java.util.Optional;
+import java.util.WeakHashMap;
+
+import org.antlr.runtime.ANTLRInputStream;
+import org.jdbi.v3.internal.SqlScriptParser;
+
+/**
+ * Locates SQL in <code>.sql</code> files on the classpath.
+ */
+public final class ClasspathSqlLocator {
+    private static final char PACKAGE_DELIMITER = '.';
+    private static final char PATH_DELIMITER = '/';
+
+    private static final SqlScriptParser SQL_SCRIPT_PARSER = new SqlScriptParser((t, sb) -> sb.append(t.getText()));
+
+    private static final Map<String, String> CACHE = synchronizedMap(new WeakHashMap<>());
+    private static final String SQL_EXTENSION = ".sql";
+
+    private ClasspathSqlLocator() {
+    }
+
+    /**
+     * Locates SQL for the given extension type and name. Example: Given an extension type <code>com.foo.Bar</code> and
+     * a name of <code>baz</code>, looks for a resource named <code>com/foo/Bar/baz.sql</code> on the classpath and
+     * returns its contents as a String.
+     *
+     * @param extensionType the extension type
+     * @param name          the SQL statement name (usually a method name of the extension type).
+     * @return the located SQL.
+     */
+    public static String findSqlOnClasspath(Class<?> extensionType, String name) {
```

Comment:
Is this really limited to extension types?  Or should we term it just as a class, same way `class.getResource()` works?